### ISSUE TITLE
Enhance Multivariate MODE to support differing numbers of forecast and observation input fields

### ISSUE THREAD

**Issue by davidalbo** [dtcenter/MET#2706]
## Describe the Enhancement ##

- Allow different number of obs and forecast inputs to create super objects, and to produce output when all the intensity flags are false.
-  Add logic to tell Mode which obs input (with intensity=TRUE) should be compared to which fcst input (with intensity = TRUE).
- `multivar_logic` and `multivar_intensity `would be parsed from the fcst and obs dictionaries rather than top-level config file context.

[Link to documentation on readthedocs](https://met.readthedocs.io/en/feature_2706_mvmode_obs_fcst_input_size/Users_Guide/mode.html#mode-tool)

### Time Estimate ###
30 hours

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
November 7 (beta2)

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Review default **alert** labels
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Milestone and Projects ###
- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**
- [ ] For the next official version, select the **MET-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose)
- dtcenter/METplus#2429 updates the METplus MODE wrapper to handle new config options.

See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y.Z Development** project for development toward the next official release
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Close this issue.

**Comment by davidalbo** [comment]
Notes taken along the way:

- Chose to use the first (0th) forecast or obs as the verification grid template, when regridding has ` "to_grid = FCST"` or  `"to_grid = OBS", `  **This needs to be documented.**
- Need to figure out what to do with some of the `'perc_thresh`' options, currently I'm disabling them when creating fcst simple objects or obs simple objects. Some of these options use two inputs.  **We should meet about this one.**
- Here and there were requirements that array lengths between obs and fcsts needed to be the same, for example the `conv_radius` array.  I've relaxed those requirements, but... currently mvmode doesn't allow multiple conv_radii.  Revisit if we ever do allow multiple conv_radii/conv_thresh and/or quilting.
- I'm replacing `multivar_intensity_flag `with comparison arrays` multivar_intensity_compare_fcst = [ 2, 1 ]; multivar_intensity_compare_obs = [2, 2];`  **This needs to be documented** **and put into the default config. and metplus as an issue**
- Did I break traditional mode with all these changes?  More likely than anything I've done so far.  **How can I test to see if I broke traditional mode?**
- Traditional mode requires number of conv thresholds = number of conv radii, why is this?  Currently mvmode requires both these numbers to be 1.  If relaxed, would it be possible that  individual obs have different values, and different fcsts have different values, and obs values not equal to fcst values?   Gets very complex  very fast.  The code WILL break down depending on what is allowed in mvmode.
